EventStore/EventStore/Service/AppointmentSchedulingEventService.cs

```diff
@@ -0,0 +1,175 @@
+﻿using EventStore.Dtos;
+using EventStore.Events;
+using EventStore.Repository;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace EventStore.Service
+{
+    public class AppointmentSchedulingEventService : IAppointmentSchedulingEventService
+    {
+        public readonly IAppointmentSchedulingEventRepository _appointmentSchedulingEventRepository;
+
+        public AppointmentSchedulingEventService(IAppointmentSchedulingEventRepository appointmentSchedulingEventRepository)
+        {
+            _appointmentSchedulingEventRepository = appointmentSchedulingEventRepository;
+        }
+
+        public DomainEvent Create(DomainEvent domainEvent)
+        {
+            return _appointmentSchedulingEventRepository.Create(domainEvent);
+        }
+
+        public List<AppointmentSchedulingEvent> GetAll()
+        {
+            return _appointmentSchedulingEventRepository.GetAll().ToList();
+        }
+
+        public long FindNextAttempt()
+        {
+            return _appointmentSchedulingEventRepository.FindNextAttempt();
+        }
+
+        private AppointmentSchedulingEvent GetEventByAttempt(long attempt)
+        {
+            return GetAll().SingleOrDefault(appointment => appointment.Attempt == attempt && appointment.EndPoint.Equals("start"));
+        }
+
+        public CountStepsEventDto GetStatisticsMinSteps()
+        {
+            if (GetSuccessfulAppointmentsAttempts().Count > 0)
+            {
+                var minEvent =GetAll().Where(appointmentEvent => GetSuccessfulAppointmentsAttempts().Contains(appointmentEvent.Attempt)).ToList().GroupBy(appointmentEvent => appointmentEvent.Attempt).Select(grp => new { Attempt = grp.Key, Count = grp.Count() - 2 }).ToList().Aggregate((eventWithMinCount, currentEvent) => currentEvent.Count <= eventWithMinCount.Count ? currentEvent : eventWithMinCount);
+                return new CountStepsEventDto(GetEventByAttempt(minEvent.Attempt), minEvent.Count);
+            }
+            return null;
+        }
+
+        public CountStepsEventDto GetStatisticsMaxSteps()
+        {
+            if (GetSuccessfulAppointmentsAttempts().Count > 0)
+            {
+                var maxEvent = GetAll().Where(appointmentEvent => GetSuccessfulAppointmentsAttempts().Contains(appointmentEvent.Attempt)).ToList().GroupBy(appointmentEvent => appointmentEvent.Attempt).Select(grp => new { Attempt = grp.Key, Count = grp.Count() - 2 }).ToList().Aggregate((eventWithMaxCount, currentEvent) => currentEvent.Count >= eventWithMaxCount.Count ? currentEvent : eventWithMaxCount);
+                return new CountStepsEventDto(GetEventByAttempt(maxEvent.Attempt), maxEvent.Count);
+            }
+            return null;
```

Comment:
Write in the form of a ternary operator.
Example:
"var maxEvent = ...;
return ..."